modules/programs/boxxy.nix

```diff
@@ -0,0 +1,110 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.programs.boxxy;
+
+  configPath = "${config.xdg.configHome}/boxxy/boxxy.yaml";
+  settingsFormat = pkgs.formats.yaml { };
+
+  boxxyRulesOpts = types.submodule {
+    freeformType = settingsFormat.type;
+
+    options = {
+      name = mkOption {
+        type = types.str;
+        description = ''
+          Unique identifier of the boxxy rule. This can be any single-line string.
+        '';
+      };
+
+      target = mkOption {
+        type = types.str;
+        default = "";
+        example = "~/.ssh";
+        description = ''
+          What directory/file to redirect.
+        '';
+      };
+
+      rewrite = mkOption {
+        type = types.str;
+        default = "";
+        example = "${xdg.configHome}/ssh";
```

Comment:
```suggestion
        example = literalExpression ''"''${config.xdg.configHome}/ssh"'';
```

Reply:
Whats up with the quotations?

Reply:
Also going to ask: Does it matter in that case for docs generation, because I didn't think that code was parsed anyway?

Reply:
If you write

```nix
example = "${xdg.configHome}/ssh";
```

you're asking nix to expand `xdg.configHome` (I guess you meant `config.xdg.configHome`) into the string, and making that the example. This will fail because you can't access `config` when building the documentation.

Instead, you want the example to be a string containing a literal nix expression, hence the weird syntax is necessary so that nix doesn't expand the `${}`. (As an alternative, this would also work: `example = "\"\${config.xdg.configHome}/ssh\"";`.)

Reply:
The code is indeed not *parsed*, but it is *evaluated* to generate the documentation (that's how you can see literal examples like `pkgs.foo` in https://nix-community.github.io/home-manager/options.html).

Reply:
So regardless where the string content is, nix was going to parse that as equal? I should've thought of that, whoops.
But yes, you're right. The end user should use ``config`` instead of the top-level variable to refer to their ``xdg.configHome``.

Reply:
> So regardless where the string content is, nix was going to parse that as equal?

Not sure what you mean by that.